saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/authentication/OpenSamlAuthenticationProvider.java

```diff
@@ -474,6 +480,33 @@ else if (subject.getEncryptedID() != null) {
 		return username;
 	}
 
+	private Map<String, Object> getAssertionAttributes(Assertion assertion) {
+		Map<String, Object> attributeMap = new HashMap<>();
+		for (AttributeStatement attributeStatement : assertion.getAttributeStatements()) {
+			for (Attribute attribute : attributeStatement.getAttributes()) {
+				List<String> attributeValues = new ArrayList<>();
+				for (XMLObject xmlObject : attribute.getAttributeValues()) {
+					attributeValues.add(getXmlObjectValue(xmlObject));
```

Comment:
It seems odd to add null values to the list. Is it necessary to preserve that aspect of the attribute statement?